lightning-invoice/src/lib.rs

```diff
@@ -1295,10 +1290,123 @@ impl<S> Display for SignOrCreationError<S> {
 	}
 }
 
+	/// Convenient utilities to create an invoice.
+pub mod utils {
+	use {Currency, Invoice, InvoiceBuilder, SemanticError, SignedRawInvoice};
+	use bech32::{FromBase32, ToBase32, u5};
+	use bitcoin_hashes::Hash;
+	use lightning::chain;
+	use lightning::chain::chaininterface::{BroadcasterInterface, FeeEstimator};
+	use lightning::chain::keysinterface::{Sign, KeysInterface};
+	use lightning::ln::channelmanager::{ChannelManager, MIN_FINAL_CLTV_EXPIRY};
+	use lightning::ln::features::InvoiceFeatures;
+	use lightning::routing::network_graph::RoutingFees;
+	use lightning::routing::router::RouteHintHop;
+	use lightning::util::logger::Logger;
+	use std::ops::Deref;
+
+	/// Utility to construct an invoice.
+	#[allow(dead_code)]
+	pub fn from_channelmanager<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>(
```

Comment:
May be useful to call this `Invoice_from_...` or so. In context I guess it makes sense but if its just in a module on its own it would be nice to reference the thing being created. Do you agree @jkczyz?

Reply:
Maybe `create_invoice_from_channelmanager` since using a `from_` prefix tends to imply a conversion.